src/main/java/picard/sam/AddOrReplaceReadGroups.java

```diff
@@ -23,10 +23,38 @@
 import picard.cmdline.programgroups.SamOrBam;
 
 import java.io.File;
-import java.util.Arrays;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * Replaces read groups in a BAM file
+ * Assigns all the reads in a file to a single new readgroup.
+ *
+ * <h3>Summary</h3>
+ * Many tools (Picard and GATK for example) require or assume the presence of at least one RG tag, defining a "readgroup"
+ * to which each read can be assigned (as specified in the RG tag in the SAM record).
+ * This tool enables the user to assign all the reads in the {@link #INPUT} to a single new readgroup.
+ * For more information about read groups, see the <a href='https://www.broadinstitute.org/gatk/guide/article?id=6472'>
+ * GATK Dictionary entry.</a>
+ * <br />
+ * This tool accepts as INPUT BAM and SAM files or URLs from the Global Alliance for Genomics and Health (GA4GH) (see http://ga4gh.org/#/documentation).
+ * <h3>Usage example:</h3>
+ * <pre>
+ * java -jar picard.jar AddOrReplaceReadGroups \\
+ *       I=input.bam \\
+ *       O=output.bam \\
+ *       RGID=4 \\
+ *       RGLB=lib1 \\
+ *       RGPL=illumina \\
+ *       RGPU=unit1 \\
+ *       RGSM=20
+ * </pre>
+ * <h3>Caveats</h3>
+ * The value of the tags must adhere (according to the <a href=\"https://samtools.github.io/hts-specs/SAMv1.pdf\"> SAM-spec</a>)
+ * with the regex '^[ -~]+$' In particular &lt;Space&gt; is the only non-printing character allowed.
```

Comment:
what is that? `<code> ... </code>` ?

Reply:
Yeah, or you could use ```{@code ...}``` which I think is easier because it escapes
HTML elements and so on, but some onliners deprecate it.   ¯\_(ツ)_/¯